### PR TITLE
Updates to Licensing & README.md

### PR DESCRIPTION
**Pull request by Rustymage**

### CODE REVIEW

File: README.md
Comment (stevequinn) [inline README.md]: While I love the detail on how fonts can be updated, can I suggest we move the font name strings ""Roboto-Regular.ttf" and "Roboto-Regular.ttf" to a .env file so the end user is not modifying code directly? This would result in:

1. A new dot.env file in the root that is committed to git that has the example .env structure:

FONTNAME="Roboto-Regular.ttf"
BOLDFONTNAME="Roboto-Medium.ttf"

2. A description in the readme installation section that tells the user they need to copy the example dot.env to .env.

3. An updated .gitignore with .env so it is not committed to git. 

4. A font section in the readme like you have that describes updating the .env vars if the user wants to change the font.

5. An updated text.py file that leverages the python-dotenv package (this will also need to be added to requirements.txt), to load the env vars. See below for a code example of this. 

```python
# !/usr/bin/env python

"""
Environment variable loader
"""

import os
from dotenv import load_dotenv

IS_LOADED = False

if not IS_LOADED:
    load_dotenv()  # take environment variables from .env.
    IS_LOADED = True

def get(name: str) -> str:
    """
    Get the value of an environment variable

    Args:
        name (str): The env name

    Returns:
        str: env value
    """
    return os.getenv(name)
```
    
    If you don't have experience with env vars I'd be happy to implement this, or if you have any aversion to the idea let me know :)
    
    Thanks.

File: LICENSE
Comment (stevequinn) [inline LICENSE:1]: Can you add the licence type header here just so it's clear? As per https://github.com/ivandokov/phockup/blob/master/license

The MIT License (MIT)

**Review by stevequinn — CHANGES_REQUESTED** [review]

File: README.md
Comment (stevequinn) [reply README.md]: Alternatively, now that I think of it, it could be moved to a parameter with appropriate bounds checking to reduce complexity:

parser.add_argument('-f', '--font', default='Roboto-Regular.ttf', help='Font file in fonts directory')
parser.add_argument('-fb', '--fontbold', default='Roboto-Medium.ttf', help='Bold font file in fonts directory')

File: LICENSE
Comment (Rustymage) [reply LICENSE:1]: Good spot, with Github picking up the license type and adding to a header I missed this. Will resolve.

File: README.md
Comment (Rustymage) [reply README.md]: Let me take a look. I did think of an environment file but with time and the need to research how to do it (my coding experience is limited, but I enjoy learning) I was going to resolve at a later date. Although, I'd be happy to pull that timeline forward and hold this PR open for a bit.

I'll have a go and see how it goes based on the guidance above. 

👍

Comment (stevequinn) [reply README.md]: I almost think parameters might be a better option? It's less initial configuration for the user, and easy to pass in new options and have sane defaults. 

What do you think?

Comment (Rustymage) [reply README.md]: So I've created it as you suggested (feels the best method) and added a validation font function to gracefully exit if the font is not available in the directory. 

I did utilise **CoPilot** for support as I got stuck on passing arguments to the main function and had it incorrectly defined. I had global definitions for font/bold font in `text.py` and that clearly wasn't working. Shifting to `main.py` under the main function fixed that with the parameters modified in `border.py`,` text.py `& `main.py`

```
font_path = os.path.join(FONTDIR, args.font)
bold_font_path = os.path.join(FONTDIR, args.fontbold)
```
Figured it out with help. The `README.md` has been updated and a small tweak to `.gitignore` too.

I won't be offended if it needs to be tidied. However, it does work as intended.

👍

File: text.py
Comment (Rustymage) [inline text.py]: Not sure `FONTINDEX` is needed, except where a single file contains multiple styles. Is this still necessary? When adding both bold and regular font files, this feels superfluous?

File: main.py
Comment (Rustymage) [inline main.py]: Validation check of font directory if argument for font is utilised.

File: main.py
Comment (Rustymage) [inline main.py]: This spits out an error depending on which font is missing when called by the argument and then gracefully exits.

File: README.md
Comment (stevequinn) [reply README.md]: Good stuff, thanks for this. I just streamlined the error reporting a little bit as there were some duplicates, and removed some redundant code. I also added python logging so we can use that instead of print statements from now on. 

I'll wait for you to review before I merge into main.

File: border.py
Comment (Rustymage) [inline border.py:132]: I see what you've done here - reduced the number of functions. Neat.

File: main.py
Comment (Rustymage) [inline main.py:60]: Agreed - I relied a little too much on CoPilot naming and wasn't consistent!

File: main.py
Comment (Rustymage) [inline main.py:81]: Can you help me understand why this is outside the main function? When I did that, it threw errors but it may have been a result of my parameters eleswhere...

File: main.py
Comment (Rustymage) [inline main.py:50]: Should this be `fontpath` & `boldfontpath` now?

File: text.py
Comment (Rustymage) [inline text.py]: I believe the removal of this is because we use the `create_font` function for both regular & bold fonts now?

Does this work if `font` and **`bold font`** are different styles?

**Review by Rustymage — COMMENTED** [review]
My review is mostly questions to educate myself and understand the improvements 👍

File: main.py
Comment (stevequinn) [reply main.py:81]: I moved it to the exif if statement as we only want to do that processing if the user has requested exif to be added, and there is actual exif data to use. 

I'm not sure why it wouldn't work for you though, as the __file__ abspath should be resolved from within any function. 

This function is turning into a bit of spaghetti code though so at some point I'll have a go at refactoring it.

File: main.py
Comment (stevequinn) [reply main.py:50]: So, it's still the font name at this  point right? The process_image function then determines the paths within the exif if statement.

File: text.py
Comment (stevequinn) [reply text.py]: That is  a good point. I'll close this PR now as it is getting quite large and covering a few bases, then how about we create another PR to add support for a font index parameter which defaults to None or 0 (need to test), and gets passed through to this create_font function?

**Pull request merged**
